checkov/terraform/tf_parser.py

```diff
@@ -605,38 +604,48 @@ def handle_variables(
                 hcl_tfvars = file
             elif file.name.endswith(".auto.tfvars.json") or file.name.endswith(".auto.tfvars"):
                 auto_vars_files.append(file)
-            elif vars_files and file.path in vars_files:
-                explicit_var_files.append(file)
             elif file.name.endswith(".tf") or file.name.endswith('.hcl'):  # TODO: add support for .tf.json
                 tf_files_to_load.append(file)
 
         for key, value in self.env_vars.items():
             if not key.startswith("TF_VAR_"):
                 continue
-            self.external_variables_data.append((key[7:], value, f"env:{key}"))
+            self.external_variables_data.append((key[7:], value, f"env:{key}", True))
```

Comment:
the variable resolution hierarchy is incorrect and should reflect the official docs https://developer.hashicorp.com/terraform/language/values/variables#variable-definition-precedence 
probably the most accurate approach would be to mark each of them with an enum value (ex. ExternalVariableType.ENV, .TFVARS, etc.) and when a new variable is read check and override an existing one if needed, then you have the up-to-date hierarchy when processing them at the end.

Reply:
In that case, I'd argue that it'd be more proper to use a dictionary and evaluate them in the correct order. If the order is truly incorrect (in which case I'd have to see why the tests that check it pass,) I'd be willing to take a stab at it in a follow-up PR.